tests/tracing/conftest.py

```diff
@@ -92,3 +95,39 @@ def databricks_tracking_uri():
 def async_logging_enabled(request, monkeypatch):
     monkeypatch.setenv(MLFLOW_ENABLE_ASYNC_LOGGING.name, str(request.param))
     return request.param
+
+
+@pytest.fixture
+def otel_collector():
+    """Start an OpenTelemetry collector in a Docker container."""
+    subprocess.run(["docker", "pull", "otel/opentelemetry-collector-contrib"], check=True)
+
+    output_file = tempfile.NamedTemporaryFile().name
+
+    with open(output_file, "w") as f:
+        process = subprocess.Popen(
+            [
+                "docker",
+                "run",
+                "-p",
+                "127.0.0.1:4317:4317",
+                "otel/opentelemetry-collector",
+            ],
+            stdout=f,
+            stderr=subprocess.STDOUT,
+            text=True,
+        )
+
+    # Wait for the collector to start
+    time.sleep(5)
```

Comment:
Is there an endpoint like `/health` that returns 200 when the collector starts running?

Reply:
Yea I wanted a health-check endpoint, but the fact that [they suggest using an extension](https://opentelemetry.io/docs/collector/configuration/#extensions) for health checking indicates it doesn't have built-in support. Let me go with this minimal approach and see if it brings any flakiness🙂